polar-core/src/formatting.rs

```diff
@@ -386,12 +386,20 @@ pub mod to_polar {
     impl ToPolarString for ExternalInstance {
         fn to_polar(&self) -> String {
             if let Some(ref repr) = self.repr {
-                repr.clone()
+                format!(
+                    "{} TYPE `{}`",
+                    repr.clone(),
+                    self.class_repr.as_ref().unwrap_or(&"UNKNOWN".to_string())
+                )
             } else {
                 // Print out external instances like ^{id: 123}
                 // NOTE: this format is used by host libraries to enrich output
                 // messages with native representations of the instances.
-                format!("^{{id: {}}}", self.instance_id)
+                format!(
+                    "^{{id: {}}} TYPE `{}`",
+                    self.instance_id,
+                    self.class_repr.as_ref().unwrap_or(&"UNKNOWN".to_string())
+                )
```

Comment:
This assumption might already be broken by the existing `ExternalInstance::to_polar` implementation, but the goal of the `ToPolarString` trait is that it should be a "reverse-parser" that turns parsed values back into valid Polar strings. If it's already broken, then I don't mind shipping this change w/ an inline comment explaining the divergence from the rest of the `ToPolarString` impls.

Reply:
On second thought, I think we should leave the `ExternalInstance::to_polar` impl exactly how it was before this PR and then create a separate function that's only used in our logging code that invokes `ExternalInstance::to_polar` and then adds the

```
 TYPE `{}`
```

suffix. What do you think?

Reply:
as discussed captured this as a follow-on notion task https://www.notion.so/osohq/Resolve-format-changes-to-ExternalInstance-to_polar-9585fe6e6216477b8179e5c215533108